nixos/modules/security/pam.nix

```diff
@@ -193,7 +202,8 @@ let
       };
 
       makeHomeDir = mkOption {
-        default = false;
+        # this needs to be enabled if GoogleOSLogin is enabled
+        default = config.security.pam.enableGoogleOSLogin;
```

Comment:
This affects all `security.pam.services.<name?>.makeHomeDir`.

Do we want to do that only for logins over ssh, not when switching to that user via different means?
Does that even make a difference, or is it more of a philosophical question?

Reply:
Honestly I'm not sure. I had the same thought as you but dismissed it as I'd think it was more surprising if things didn't work the same across services.

Reply:
This was changed.

Setting `security.googleOsLogin.enable` to true will now set
```
security.pam.services.sshd.makeHomeDir = true
security.pam.services.sshd.googleOsLoginAccountVerification = true
```

And we'll add a trigger for setting `googleOsLoginAuthentication`. This removes some oslogin specific cruft from the PAM module itself, and only configures oslogin pam for sshd.